### PR TITLE
acc, set, applet_oe: stub various functions, add set service

### PR DESCRIPTION
**Pull request by goaaats**

### CODE REVIEW

File: src/core/hle/service/set/set.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class SET final : public ServiceFramework<SET> {
+public:
+    explicit SET(const char* name);
+    ~SET() = default;
+
+private:
+    void GetAvailableLanguageCodes(Kernel::HLERequestContext& ctx);
+};
+
+/// Registers all Time services with the specified service manager.
```
Comment (Rozelette) [inline src/core/hle/service/set/set.h]: "Time" copy/paste error?

File: src/core/hle/service/am/applet_oe.cpp [hunk 2]
```diff
@@ -314,6 +315,13 @@ class IApplicationFunctions final : public ServiceFramework<IApplicationFunction
         LOG_WARNING(Service, "(STUBBED) called, result=0x%08X", result);
     }
 
+    void GetDesiredLanguage(Kernel::HLERequestContext& ctx) {
+        IPC::RequestBuilder rb{ctx, 2};
```
Comment (Rozelette) [inline src/core/hle/service/am/applet_oe.cpp]: `rb{ctx, 4};` (+2 words from u64)

File: src/core/CMakeLists.txt [hunk 3]
```diff
@@ -139,6 +139,8 @@ add_library(core STATIC
     hle/service/vi/vi.h
     hle/service/vi/vi_m.cpp
     hle/service/vi/vi_m.h
+    hle/service/set/set.cpp
```
Comment (Rozelette) [inline src/core/CMakeLists.txt]: Files should be in alphabetical order

File: src/core/hle/service/set/set.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+#include "core/hle/ipc_helpers.h"
+#include "core/hle/kernel/client_port.h"
+#include "core/hle/kernel/client_session.h"
+#include "core/hle/service/set/set.h"
+
+namespace Service {
+namespace Set {
+
+void SET::GetAvailableLanguageCodes(Kernel::HLERequestContext& ctx) {
+    constexpr u8 data[13] = {
```
Comment (lioncash) [inline src/core/hle/service/set/set.cpp]:
```cpp
constexpr std::array<u8, 13> data{{
    0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
}};
```

**Comment by Subv** [comment]
Please squash all those fixup commits into a single one

File: src/core/hle/service/set/set.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+#include "core/hle/ipc_helpers.h"
+#include "core/hle/kernel/client_port.h"
+#include "core/hle/kernel/client_session.h"
+#include "core/hle/service/set/set.h"
+
+namespace Service {
+namespace Set {
+
+void SET::GetAvailableLanguageCodes(Kernel::HLERequestContext& ctx) {
+    constexpr std::array<u8, 13> langCodes{{
```
Comment (Subv) [inline src/core/hle/service/set/set.cpp]: Our codestyle is to use lower_snake_case for variables. `lang_codes`

**Comment by goaaats** [comment]
Squashed and fixed casing

File: src/core/hle/service/acc/acc_u0.cpp [hunk 6]
```diff
@@ -9,15 +9,36 @@
...
+void ACC_U0::GetUserExistence(Kernel::HLERequestContext& ctx) {
+    LOG_WARNING(Service, "(STUBBED) called");
+    IPC::RequestBuilder rb{ctx, 2};
+    rb.Push(RESULT_SUCCESS);
+}
+
+void ACC_U0::GetProfile(Kernel::HLERequestContext& ctx) {
+    LOG_WARNING(Service, "(STUBBED) called");
+    IPC::RequestBuilder rb{ctx, 2};
+    rb.Push(RESULT_SUCCESS);
```
Comment (bunnei) [inline src/core/hle/service/acc/acc_u0.cpp]: This returns an IProfile interface, see https://reswitched.github.io/SwIPC/ifaces.html#nn::account::IAccountServiceForApplication(5)

File: src/core/hle/service/acc/acc_u0.cpp [hunk 7]
```diff
@@ -9,15 +9,36 @@
 namespace Service {
 namespace Account {
 
+void ACC_U0::GetUserExistence(Kernel::HLERequestContext& ctx) {
+    LOG_WARNING(Service, "(STUBBED) called");
+    IPC::RequestBuilder rb{ctx, 2};
+    rb.Push(RESULT_SUCCESS);
```
Comment (bunnei) [inline src/core/hle/service/acc/acc_u0.cpp]: This returns a bool, see:
https://reswitched.github.io/SwIPC/ifaces.html#nn::account::IAccountServiceForApplication(1)

File: src/core/hle/service/acc/acc_u0.cpp [hunk 8]
```diff
@@ -9,15 +9,36 @@
...
 void ACC_U0::InitializeApplicationInfo(Kernel::HLERequestContext& ctx) {
     LOG_WARNING(Service, "(STUBBED) called");
     IPC::RequestBuilder rb{ctx, 2};
     rb.Push(RESULT_SUCCESS);
 }
 
+void ACC_U0::GetBaasAccountManagerForApplication(Kernel::HLERequestContext& ctx) {
+    LOG_WARNING(Service, "(STUBBED) called");
+    IPC::RequestBuilder rb{ctx, 2};
+    rb.Push(RESULT_SUCCESS);
```
Comment (bunnei) [inline src/core/hle/service/acc/acc_u0.cpp]: This returns an IManagerForApplication interface, see
https://reswitched.github.io/SwIPC/ifaces.html#nn::account::IAccountServiceForApplication(101)

File: src/core/hle/service/am/applet_oe.cpp [hunk 9]
```diff
@@ -314,6 +315,13 @@ class IApplicationFunctions final : public ServiceFramework<IApplicationFunction
         LOG_WARNING(Service, "(STUBBED) called, result=0x%08X", result);
     }
 
+    void GetDesiredLanguage(Kernel::HLERequestContext& ctx) {
+        IPC::RequestBuilder rb{ctx, 4};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u64>(1); // English?
```
Comment (bunnei) [inline src/core/hle/service/am/applet_oe.cpp]: Please put a TODO here explaining that this value needs to be verified on HW

File: src/core/hle/service/set/set.cpp [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+#include "core/hle/ipc_helpers.h"
+#include "core/hle/kernel/client_port.h"
+#include "core/hle/kernel/client_session.h"
+#include "core/hle/service/set/set.h"
+
+namespace Service {
+namespace Set {
+
+void SET::GetAvailableLanguageCodes(Kernel::HLERequestContext& ctx) {
+    constexpr std::array<u8, 13> lang_codes{{
```
Comment (bunnei) [inline src/core/hle/service/set/set.cpp]: `std::array` gets initialized with zeros by default

File: src/core/hle/service/am/applet_oe.cpp (hunk 9 above)
Comment (gdkchan) [reply src/core/hle/service/am/applet_oe.cpp]: 0 is Japanese and 1 is English, the other languages still needs to be verified. I think its a good idea to create a enum for those and use the value from the enum -- this eliminates the need for the comment too.

**Comment by bunnei** [comment]
Please fix clang-format errors

File: src/core/hle/service/acc/acc_u0.cpp [hunk 11]
```diff
@@ -9,15 +9,85 @@
...
+        UserData user_data{};
+        IPC::RequestBuilder rb{ctx, 11};
+        rb.Push(RESULT_SUCCESS);
+        rb.PushRaw(user_data);
+    }
+
+    void GetBase(Kernel::HLERequestContext& ctx) {
+        LOG_WARNING(Service, "(STUBBED) called");
+        ProfileBase profile_base{};
+        IPC::RequestBuilder rb{ctx, 11};
```
Comment (bunnei) [inline src/core/hle/service/acc/acc_u0.cpp]: should be: `(sizeof(ProfileBase) / 4) + 2`
(size of ProfileBase in words, + 2 words for result code)

File: src/core/hle/service/acc/acc_u0.cpp [hunk 12]
```diff
@@ -9,15 +9,85 @@
...
+            {1, &IProfile::GetBase, "GetBase"},
+        };
+        RegisterHandlers(functions);
+    }
+
+private:
+    void Get(Kernel::HLERequestContext& ctx) {
+        LOG_WARNING(Service, "(STUBBED) called");
+        UserData user_data{};
+        IPC::RequestBuilder rb{ctx, 11};
```
Comment (bunnei) [inline src/core/hle/service/acc/acc_u0.cpp]: Is this right? See https://reswitched.github.io/SwIPC/ifaces.html#nn::account::profile::IProfile(0)
Also, please derive the size from the data you are returning, e.g. see my comment below

**Pull request merged**